mixer/pkg/il/testing/tests.go

```diff
@@ -3615,6 +3615,254 @@ end
 			"ab": false,
 		},
 	},
+	{
+		E:    `as + bs`,
+		Type: descriptor.STRING,
+		R:    "ab",
+		I: map[string]interface{}{
+			"as": "a",
+			"bs": "b",
+		},
+		IL: `
+fn eval() string
+  resolve_s "as"
+  resolve_s "bs"
+  add_s
+  ret
+end
+`,
+	},
+	{
+		E:    `as + "b"`,
+		Type: descriptor.STRING,
+		R:    "ab",
+		I: map[string]interface{}{
+			"as": "a",
+		},
+		IL: `
+fn eval() string
+  resolve_s "as"
+  apush_s "b"
+  add_s
+  ret
+end
+`,
+	},
+	{
+		E:    `"b" + as`,
+		Type: descriptor.STRING,
+		R:    "ba",
+		I: map[string]interface{}{
+			"as": "a",
+		},
+		IL: `
+fn eval() string
+  apush_s "b"
+  resolve_s "as"
+  add_s
+  ret
+end
+`,
+	},
+	{
+		E:    `"a" + "b"`,
+		Type: descriptor.STRING,
+		R:    "ab",
+		IL: `
+fn eval() string
+  apush_s "a"
+  apush_s "b"
+  add_s
+  ret
+end
+`,
+	},
+	{
+		E:    `as | "unknown" + "b"`,
+		Type: descriptor.STRING,
+		R:    "ab",
+		I: map[string]interface{}{
+			"as": "a",
+		},
+		IL: `
+fn eval() string
+  tresolve_s "as"
+  jnz L0
+  apush_s "unknown"
+L0:
+  apush_s "b"
+  add_s
+  ret
+end
+`,
+	},
+	{
+		E:    `as | "unknown" + "b"`,
+		Type: descriptor.STRING,
+		R:    "ab",
+		I: map[string]interface{}{
+			"as": "a",
```

Comment:
It looks like you meant this for the "unknownb" case (it is a duplicate of the one above).